cfme/common/vm.py

```diff
@@ -483,7 +483,7 @@ def create(self, vm_name, provider, form_values=None, cancel=False,
         else:
             inst_args = vm.vm_default_args
             form_values = recursive_update(inst_args, form_values)
-        if form_values.get('environment').get('automatic_placement'):
+        if form_values.get('environment', {}).get('automatic_placement'):
```

Comment:
This will still fail if ``environment`` happens to be ``None`` (e.g. when constructed using method that returns dictionary or ``None``).

How about
```python
environment = form_values.get('environment') or {}
if environment.get('automatic_placement'):
    ...
```
?

Reply:
Added